### DISCUSSION TITLE
Build -> Managed-lern -> Ml-Report-Service  (Build job Failing in Dockerfile instructions)

### DISCUSSION THREAD

**Discussion by srikanth-6520** [Sunbird-Ed/Community#497] [Bugs] [answered]
Hello @gandham-santhosh ,we have tried to Build ml-report-service with tag "release-5.1.0_RC4",

Jenkins-Logs:

Started by user priyap
Rebuilds build #[2](https://jenkins.alex-sb.dev.verlab.io/job/Build/job/managed-learn/job/ml-report-service/2)
Checking out git https://github.com/project-sunbird/ml-reports-service.git into /var/lib/jenkins/workspace/Build/managed-learn/ml-report-service@script/a66c09b5951168a8764237aaab7b97269032d3faf673d217cb04379862d795f4 to read Jenkinsfile
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Cloning repository https://github.com/project-sunbird/ml-reports-service.git
 > git init /var/lib/jenkins/workspace/Build/managed-learn/ml-report-service@script/a66c09b5951168a8764237aaab7b97269032d3faf673d217cb04379862d795f4 # timeout=10
Fetching upstream changes from https://github.com/project-sunbird/ml-reports-service.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress -- https://github.com/project-sunbird/ml-reports-service.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/project-sunbird/ml-reports-service.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 > git rev-parse origin/release-5.1.0_RC4^{commit} # timeout=10
 > git rev-parse release-5.1.0_RC4^{commit} # timeout=10
Checking out Revision 76a9c4d3cce8e0d8b912da6beb3f533846ff1e44 (release-5.1.0_RC4)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 76a9c4d3cce8e0d8b912da6beb3f533846ff1e44 # timeout=10
Commit message: "Merge pull request #81 from shikshalokam/master"
 > git rev-list --no-walk 76a9c4d3cce8e0d8b912da6beb3f533846ff1e44 # timeout=10
[Pipeline] Start of Pipeline
[Pipeline] node
Running on [Jenkins](https://jenkins.alex-sb.dev.verlab.io/computer/(built-in)/) in /var/lib/jenkins/workspace/Build/managed-learn/ml-report-service
[Pipeline] {
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] timestamps
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Checkout)
[Pipeline] echo
09:58:15  Found environment variable named hub_org with value as: 961571523562.dkr.ecr.ap-south-1.amazonaws.com/prod
[Pipeline] }
[Pipeline] // stage
[Pipeline] checkout
09:58:15  The recommended git tool is: NONE
09:58:15  No credentials specified
09:58:15  Cloning the remote Git repository
09:58:15  Cloning repository https://github.com/project-sunbird/ml-reports-service.git
09:58:15   > git init /var/lib/jenkins/workspace/Build/managed-learn/ml-report-service # timeout=10
09:58:15  Fetching upstream changes from https://github.com/project-sunbird/ml-reports-service.git
09:58:15   > git --version # timeout=10
09:58:15   > git --version # 'git version 2.17.1'
09:58:15   > git fetch --tags --progress -- https://github.com/project-sunbird/ml-reports-service.git +refs/heads/*:refs/remotes/origin/* # timeout=10
09:58:16   > git config remote.origin.url https://github.com/project-sunbird/ml-reports-service.git # timeout=10
09:58:16   > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
09:58:16  Avoid second fetch
09:58:16   > git rev-parse origin/release-5.1.0_RC4^{commit} # timeout=10
09:58:16   > git rev-parse release-5.1.0_RC4^{commit} # timeout=10
09:58:16  Checking out Revision 76a9c4d3cce8e0d8b912da6beb3f533846ff1e44 (release-5.1.0_RC4)
09:58:16   > git config core.sparsecheckout # timeout=10
09:58:16   > git checkout -f 76a9c4d3cce8e0d8b912da6beb3f533846ff1e44 # timeout=10
09:58:16  Commit message: "Merge pull request #81 from shikshalokam/master"
[Pipeline] sh
09:58:16  + git rev-parse --short HEAD
[Pipeline] sh
09:58:16  + echo release-5.1.0_RC4_76a9c4d_3
[Pipeline] echo
09:58:16  build_tag: release-5.1.0_RC4_76a9c4d_3
[Pipeline] }
[Pipeline] // timestamps
[Pipeline] stage
[Pipeline] { (Build)
[Pipeline] echo
Environment will be : build
[Pipeline] sh
+ git submodule update --init
[Pipeline] sh
+ git submodule update --init --recursive --remote
[Pipeline] sh
+ chmod 777 build.sh
[Pipeline] sh
+ ./build.sh release-5.1.0_RC4_76a9c4d_3 built-in 961571523562.dkr.ecr.ap-south-1.amazonaws.com/prod
#1 [internal] load .dockerignore
#1 transferring context: 79B done
#1 DONE 0.0s

#2 [internal] load build definition from Dockerfile
#2 transferring dockerfile: 371B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/library/node:12
#3 DONE 1.5s

#4 [1/6] FROM docker.io/library/node:12@sha256:01627afeb110b3054ba4a1405541ca095c8bfca1cb6f2be9479c767a2711879e
#4 DONE 0.0s

#5 [internal] load build context
#5 transferring context: 3.97MB 0.1s done
#5 DONE 0.1s

#6 [2/6] WORKDIR /opt/reports
#6 CACHED

#7 [3/6] COPY package.json /opt/reports
#7 CACHED

#8 [4/6] RUN apt update && apt install fonts-indic -y     && fc-cache -f
#8 0.408 
#8 0.408 WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
#8 0.408 
#8 0.469 Ign:1 http://deb.debian.org/debian stretch InRelease
#8 0.469 Ign:2 http://security.debian.org/debian-security stretch/updates InRelease
#8 0.471 Ign:3 http://deb.debian.org/debian stretch-updates InRelease
#8 0.471 Err:4 http://security.debian.org/debian-security stretch/updates Release
#8 0.471   404  Not Found
#8 0.473 Err:5 http://deb.debian.org/debian stretch Release
#8 0.473   404  Not Found
#8 0.475 Err:6 http://deb.debian.org/debian stretch-updates Release
#8 0.475   404  Not Found
#8 0.489 Reading package lists...
#8 0.502 E: The repository 'http://security.debian.org/debian-security stretch/updates Release' does not have a Release file.
#8 0.502 E: The repository 'http://deb.debian.org/debian stretch Release' does not have a Release file.
#8 0.502 E: The repository 'http://deb.debian.org/debian stretch-updates Release' does not have a Release file.
#8 ERROR: process "/bin/sh -c apt update && apt install fonts-indic -y     && fc-cache -f" did not complete successfully: exit code: 100
------
 > [4/6] RUN apt update && apt install fonts-indic -y     && fc-cache -f:
#8 0.471 Err:4 http://security.debian.org/debian-security stretch/updates Release
#8 0.471   404  Not Found
#8 0.473 Err:5 http://deb.debian.org/debian stretch Release
#8 0.473   404  Not Found
#8 0.475 Err:6 http://deb.debian.org/debian stretch-updates Release
#8 0.475   404  Not Found
#8 0.489 Reading package lists...
#8 0.502 E: The repository 'http://security.debian.org/debian-security stretch/updates Release' does not have a Release file.
#8 0.502 E: The repository 'http://deb.debian.org/debian stretch Release' does not have a Release file.
#8 0.502 E: The repository 'http://deb.debian.org/debian stretch-updates Release' does not have a Release file.
------
Dockerfile:9
--------------------
   8 |     #install fonts
   9 | >>> RUN apt update && apt install fonts-indic -y \
  10 | >>>     && fc-cache -f 
  11 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c apt update && apt install fonts-indic -y     && fc-cache -f" did not complete successfully: exit code: 100
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 1
Finished: FAILURE

**Comment by santhosh-tg** [comment]
@aks30 @VISHNUDAS-tunerlabs Could you please advise on this issue

**Reply by aks30** [reply, marked as answer]
This is being fixed in the 6.0 release, however, if its a blocker, for now, you can update the node version in Dockerfile to 16. Refer here - https://github.com/project-sunbird/ml-reports-service/blob/release-6.0.0/Dockerfile

**Reply by chitranshu-keshav** [reply]
@aks30 can you please add the JIRA ticket here and mark the thread as answered

**Reply by VISHNUDAS-tunerlabs** [reply]
@chitranshu-keshav :  https://project-sunbird.atlassian.net/browse/ED-2681 Marking this thread as Answered.
